c/CMakeLists.txt

```
@@ -1,74 +1,141 @@
-  set_target_properties(blake3-static PROPERTIES OUTPUT_NAME blake3)
-  set_target_properties(blake3-static PROPERTIES VERSION ${PROJECT_VERSION})
-  set_target_properties(blake3-static PROPERTIES SOVERSION 0)
-  set_target_properties(blake3-static PROPERTIES PUBLIC_HEADER blake3.h)
-  set_target_properties(blake3-static PROPERTIES COMPILE_FLAGS -fPIC)
-  target_include_directories(blake3-static PUBLIC .)
+elseif((CMAKE_SYSTEM_PROCESSOR STREQUAL "i686"
+          OR CMAKE_SYSTEM_PROCESSOR STREQUAL "X86")
+       AND DEFINED BLAKE3_CFLAGS_SSE2
+       AND DEFINED BLAKE3_CFLAGS_SSE4.1
+       AND DEFINED BLAKE3_CFLAGS_AVX2
+       AND DEFINED BLAKE3_CFLAGS_AVX512)
+  target_sources(blake3 PRIVATE
+    blake3_avx2.c
+    blake3_avx512.c
+    blake3_sse2.c
+    blake3_sse41.c
+  )
+  set_source_files_properties(blake3_avx2.c PROPERTIES COMPILE_FLAGS "${BLAKE3_CFLAGS_AVX2}")
+  set_source_files_properties(blake3_avx512.c PROPERTIES COMPILE_FLAGS "${BLAKE3_CFLAGS_AVX512}")
+  set_source_files_properties(blake3_sse2.c PROPERTIES COMPILE_FLAGS "${BLAKE3_CFLAGS_SSE2}")
+  set_source_files_properties(blake3_sse41.c PROPERTIES COMPILE_FLAGS "${BLAKE3_CFLAGS_SSE4.1}")
 
-  install(TARGETS blake3-static
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+elseif((ANDROID_ABI STREQUAL "armeabi-v7a"
+          OR CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64"
+          OR CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64") # For M1 macs
+       AND DEFINED BLAKE3_CFLAGS_NEON)
+  target_sources(blake3 PRIVATE
+    blake3_neon.c
+  )
+  set_source_files_properties(blake3_dispatch.c PROPERTIES COMPILE_DEFINITIONS BLAKE3_USE_NEON=1)
+  set_source_files_properties(blake3_neon.c PROPERTIES COMPILE_FLAGS "${BLAKE3_CFLAGS_NEON}")
+endif()
 
-else()
+# cmake install support
+install(FILES blake3.h DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
+install(TARGETS blake3 EXPORT blake3-targets)
+install(EXPORT blake3-targets
+  NAMESPACE BLAKE3::
+  DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/blake3"
+)
 
-  add_library(blake3-shared SHARED ${blake3_SOURCES})
-  set_target_properties(blake3-shared PROPERTIES VERSION ${PROJECT_VERSION})
-  set_target_properties(blake3-shared PROPERTIES SOVERSION 0)
-  set_target_properties(blake3-shared PROPERTIES PUBLIC_HEADER blake3.h)
-  target_include_directories(blake3-shared PUBLIC .)
-  
-  install(TARGETS blake3-shared
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+include(CMakePackageConfigHelpers)
+configure_package_config_file(blake3-config.cmake.in
+    "${CMAKE_CURRENT_BINARY_DIR}/blake3-config.cmake"
 
-endif(BLAKE3_STATIC)
+    INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/blake3"
+)
+write_basic_package_version_file(
+    "${CMAKE_CURRENT_BINARY_DIR}/blake3-config-version.cmake"
+    VERSION ${libblake3_VERSION}
+    COMPATIBILITY SameMinorVersion # as long as libblake3_VERSION < 1.0.0
```

```suggestion
    COMPATIBILITY SameMajorVersion
```
If the previous suggestion wrt the version is accepted, this ought be the correct policy.